hydra-main/src/main/java/com/addthis/hydra/job/web/SearchResource.java

```diff
@@ -0,0 +1,49 @@
+package com.addthis.hydra.job.web;
+
+import com.addthis.hydra.job.spawn.Spawn;
+import com.addthis.hydra.job.spawn.search.SearchOptions;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+
+@Path("/search")
+public class SearchResource {
+    private final Spawn spawn;
+
+    public SearchResource(Spawn spawn) {
+        this.spawn = spawn;
+    }
+
+    @GET
+    @Path("/all")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getSearch(@QueryParam("q") String q) {
+        try {
+            PipedInputStream results = spawn.getSearchResultStream(new SearchOptions(q));
```

Comment:
Sorry the comment was misleading. You are using this variable inside a nested inner class not a lambda. But the same principle applies in fact I think the JVM uses the same mechanism to implement both features.